File: BookWorm/BookWorm.Services/DataBase/BookReview.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookWorm.Services.DataBase
{
    public class BookReview
    {
        [Key]
        public int Id { get; set; }
        
        public int UserId { get; set; }

        [ForeignKey("UserId")]
        public User User { get; set; } = null!;
        
        public int BookId { get; set; }

        [ForeignKey("BookId")]
        public Book Book { get; set; } = null!;
        
        [MaxLength(2000)]
        public string? Review { get; set; } = string.Empty;

        [Range(1, 5)]
        public int Rating { get; set; }
        public bool isChecked { get; set; } = false;

        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
} 

File: BookWorm/UI/bookworm_desktop/lib/model/bookReview.dart
import 'package:json_annotation/json_annotation.dart';

part 'bookReview.g.dart';
@JsonSerializable()
class BookReview {
  final int id;
  final int userId;
  final String userName;
  final int bookId;
  final String bookTitle;
  final String? review;
  final int rating;
  final bool isChecked;
  final DateTime createdAt;


  BookReview({
    this.id = 0,
    this.userId = 0,
    this.userName = '',
    this.bookId = 0,
    this.bookTitle = '',
    this.review = '',
    this.rating = 0,
    this.isChecked = false,
    DateTime? createdAt,
  }) : createdAt = createdAt ?? DateTime.now();
  factory BookReview.fromJson(Map<String, dynamic> json) => _$BookReviewFromJson(json);
  
  }
